### PR TITLE
feat: Add `acceptedAt` for payments

### PR DESCRIPTION
**Pull request by warmwaffles**
Adds `acceptedAt` to `Payment` and `Credit` objects.

### CODE REVIEW

File: vic.api.v0.yaml [hunk 1]
```diff
@@ -2305,8 +2305,21 @@ components:
...
+                withdrawn. This is the date the payments were submitted to the
+                bank and were accepted by the bank.
+            - type: 'null'
+        acceptedAt:
+          oneOf:
+            - type: string
+              format: date-time
+              description: |
+                The time the payment processor has accepted the payment request
+                and is currently processing the payment.
```
Comment (jonathanpglick) [inline vic.api.v0.yaml:2322]:
```suggestion
                The date time the payment request was submitted and accepted
                for processing.
```

File: vic.api.v0.yaml [hunk 2]
```diff
@@ -2381,11 +2394,21 @@ components:
...
               format: date-time
               description: |
-                A rough date time when the funds were withdrawn.
+                A rough date time when the funds were withdrawn. There is no
+                guarantee on when this will be set. The times can vary between
+                15 minutes to 4 hours and beyond. If you need a time when the
+                payment was submitted, see the `acceptedAt` field.
                 **Note**: This is not an exact date time of when the funds were
                 withdrawn. This is the date the payments were submitted to the
                 bank and were accepted by the bank.
```
Comment (jonathanpglick) [inline vic.api.v0.yaml:2403]:
```suggestion
                The date time the payment request began to be processed.
                
                **Note**: This is not the actual date time that the funds were
                withdrawn from the bank. The payment request can take 15
                minutes to 4 hours or more to be processed. See the `acceptedAt`
                field for the date time that the payment was submitted and
                accepted.
```

File: vic.api.v0.yaml [hunk 3]
```diff
@@ -2381,11 +2394,21 @@ components:
...
             - type: 'null'
+        acceptedAt:
+          oneOf:
+            - type: string
+              format: date-time
+              description: |
+                The time the payment processor has accepted the payment request
+                and is currently processing the payment.
```
Comment (jonathanpglick) [inline vic.api.v0.yaml:2411]:
```suggestion
                The date time the payment request was submitted and accepted
                for processing.
            - type: 'null'
```

**Review by jonathanpglick — COMMENTED** [review]
@warmwaffles I took at stab at it. Thanks for chatting the context so I could understand this! What do you think?

I stumbled on the references to the "payment processor" and the "bank" so I phrased it so that the details of the processor and the bank weren't needed.

It seems like these objects abstract the complexity of the processor and bank interactions so we don't need to describe them here. From what I understand, we have:

1. The date time the payment was submitted and accepted: `acceptedAt`
2. The date time the submitted and accepted payment started processing: `fundedAt`

Also, I realize I used "payment request" in these instead of "payment". It felt more natural for the context. But are these descriptions actually talking about the `Payment` object that's in these docs? Or a more conceptual idea of a "payment"?

**Comment by warmwaffles** [comment]
> are these descriptions actually talking about the `Payment` object that's in these docs?

Yes it is talking about the `Payment` and `Credit`. Although, `Credit`s are not "funded", they are applied against the balance owed.

**Pull request merged**